addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2098,8 +2101,11 @@ var SnippetsMenu = Widget.extend({
      */
     _onRemoveSnippet: function (ev) {
         ev.stopPropagation();
-        this._createSnippetEditor(ev.data.$snippet).then(function (editor) {
-            editor.removeSnippet();
+        this._createSnippetEditor(ev.data.$snippet).then(async editor => {
+            await editor.removeSnippet();
```

Comment:
@sdegueldre Note: in fact that method is not yet async but will be if I merge this: https://github.com/odoo/odoo/pull/43918 ?

Reply:
There's been a lot of code shuffling between this and the column number fix so I must've lost that somewhere. Same can be said for the `_onCloneSnippet` method, I made both of those async at some point but the code got lost somewhere. I can make both of those async and take the `onSuccess` param on the event in the first commit we discussed. As a side note, I've found myself adding this mechanism to events quite a few times, and I've seen it done by existing code quite a bit too. Might want to consider baking it into the event system at some point?

Reply:
Not sure what you would bake. Mutex-alize event handlers calls automatically per widget? Dangerous because you will not be able to preventDefault / stopPropagation correctly/intuitively. If the async part of an event handler must be waited, it seems better to make it explicit for now. And for some cases the 'async' word in the event declaration is enough to make sure a specific handler is "mutex-alized" with himself.

In any case, no point in trying to improve the old framework. The next step at improving widgets is probably using owl on our side (at least in the editor).